src/App.jsx

```diff
@@ -14,45 +17,125 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ABC = 'abc';
+const SORT_FIELD_LENGTH = 'length';
+let REVERSE = false;
+let ABC = false;
+let LENGTH = false;
+
+export const App = () => {
+  let visibleGoods = [...goodsFromServer];
+  const [count, setCount] = useState(0);
+  const [sortField, setSortField] = useState('');
+  const hasResetButton = REVERSE === true || sortField !== '';
+
+  if (count % 2 !== 0) {
+    if (sortField) {
+      visibleGoods.sort((good1, good2) => {
+        switch (sortField) {
+          case SORT_FIELD_ABC:
+            return good2.localeCompare(good1);
+
+          case SORT_FIELD_LENGTH:
+            return good1.length - good2.length;
+
+          default:
+            return 0;
+        }
+      });
+    }
+
+    if (!sortField) {
+      visibleGoods.reverse();
+    }
+  } else {
+    if (sortField) {
+      visibleGoods.sort((good1, good2) => {
+        switch (sortField) {
+          case SORT_FIELD_ABC:
+            return good1.localeCompare(good2);
+
+          case SORT_FIELD_LENGTH:
+            return good2.length - good1.length;
+
+          default:
+            return 0;
+        }
+      });
+    }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+    if (!sortField) {
+      visibleGoods = [...goodsFromServer];
+    }
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': ABC === false,
+          })}
+          onClick={() => {
+            ABC = true;
+            LENGTH = false;
+            setSortField(SORT_FIELD_ABC);
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': LENGTH === false,
+          })}
+          onClick={() => {
+            ABC = false;
+            LENGTH = true;
+            setSortField(SORT_FIELD_LENGTH);
+          }}
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**